src/main/java/com/amazon/ion/impl/macro/MacroEvaluatorAsIonReader.kt

```diff
@@ -56,7 +56,10 @@ class MacroEvaluatorAsIonReader(
     }
 
     override fun next(): IonType? {
-        if (!hasNext()) return null
+        if (!hasNext()) {
+            currentValueExpression = null
```

Comment:
This resets the reader between expressions; without this, `getType()` will return non-null after the end of a container is reached.